tests/unit/repo/experiments/queue/test_local.py

```diff
@@ -35,3 +35,35 @@ def test_shutdown_active_tasks(test_queue, mocker):
     assert "foo" == result.get()
     assert result.id not in test_queue._shutdown_task_ids
     shutdown_spy.assert_called_once()
+
+
+def test_post_run_after_kill(test_queue):
+    import time
+
+    from celery import chain
+
+    sig_bar = test_queue.proc.run(
+        ["python3", "-c", "import time; time.sleep(5)"], name="bar"
+    )
+    result_bar = sig_bar.freeze()
+    sig_foo = _foo.s()
+    result_foo = sig_foo.freeze()
+    run_chain = chain(sig_bar, sig_foo)
+
+    run_chain.delay()
+    timeout = time.time() + 10
+
+    while True:
+        if result_bar.status == "RUNNING":
+            break
+        if time.time() > timeout:
+            raise AssertionError()
+
+    assert result_foo.status == "PENDING"
+    test_queue.kill("bar")
+
+    while True:
+        if result_foo.status == "SUCCESS":
```

Comment:
Here, even if we killed the process "bar", the `result_bar.status` still returns "SUCCESS", only  `celery.control.revoke` would make it fail, and this is why the following "foo" process will continue.